File: hackerrank/easy/0008-kangaroo/js/solution.mjs
// solution.mjs

export function kangaroo(x1, v1, x2, v2) {
  if (v1 === v2) return x1 === x2 ? "YES" : "NO";
  const dx = x2 - x1;
  const dv = v1 - v2;
  // divisible and n >= 0
  return dx % dv === 0 && dx / dv >= 0 ? "YES" : "NO";
}

// --- demo run ---
console.log(kangaroo(0, 3, 4, 2)); // YES
console.log(kangaroo(0, 2, 5, 3)); // NO

File: hackerrank/easy/0008-kangaroo/python/solution.py
## solution.py

def kangaroo(x1: int, v1: int, x2: int, v2: int) -> str:
    if v1 == v2:
        return "YES" if x1 == x2 else "NO"
    dx = x2 - x1
    dv = v1 - v2
    return "YES" if dx % dv == 0 and dx // dv >= 0 else "NO"


# --- demo run ---
if __name__ == "__main__":
    print(kangaroo(0, 3, 4, 2))  # YES
    print(kangaroo(0, 2, 5, 3))  # NO